src/main/java/pong/Ball.java

```diff
@@ -1,4 +1,65 @@
 package pong;
 
 public class Ball {
+    private Paddle userPaddle;
+    private Paddle compPaddle;
+
+    private double x;
+    private double y;
+    private double velX;
+    private double velY;
+    private final int MID_OF_BALL = PongView.BALL_WIDTH / 2;
+
+    public Ball(Paddle userPaddle, Paddle compPaddle) {
+        //first position - middle of the board.
+        x = Environment.WIDTH / 2;
+        y = Environment.HEIGHT / 2;
+
+        //random velocity to start the game off
+        velY = getRandomVelocity();
+        velX = getRandomVelocity();
+        this.userPaddle = userPaddle;
+        this.compPaddle = compPaddle;
+    }
+
+    private double getRandomVelocity() {
+        //randSpeed can be any number 2-4
+        double randSpeed = (Math.random() * 3 + 2);
+        double randDirection = Math.random() * 2;
+        //randDirection will be assigned as either 1 or -1
+        randDirection = randDirection == 1? 1: -1;
```

Comment:
when I test this out, I only get -1

Reply:
Oooh, thanks for testing. I'll look into this

Reply:
Fixed this in commit 8d07d6. It was an because I hadn't taken into account that Math.random returns a double. I probably could also instead use a random object instead...